compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELogicalObjFifoSplittingUtils.cpp

```diff
@@ -351,31 +375,13 @@ LogicalResult splitLogicalObjectFifos(
   SmallVector<OpFoldResult> staticL3AsSourceSizes =
       l3ToL2DmaOp.getSourceMixedSizes();
 
-  LogicalObjectFifoFromMemrefOp l2TargetObjectFifo =
-      l3ToL2DmaOp.getTargetObjectFifo();
-  ArrayRef<int64_t> l2TargetShape =
-      l2TargetObjectFifo.getMemrefType().getShape();
-  if (l2TargetShape.size() != staticL2AsTargetSizes.size()) {
-    LLVM_DEBUG(llvm::dbgs() << "L2 target size should be the same");
-    return failure();
-  }
-
-  // Check if the L3->L2 dma is transposed on the target side.
-  bool dmaTransposeOnSource = true;
-  for (auto [s1, s2] : llvm::zip_equal(l2TargetShape, staticL2AsTargetSizes)) {
-    if (s1 != getConstantIntValue(s2)) {
-      dmaTransposeOnSource = false;
-      break;
-    }
-  }
-  if (staticL3AsSourceSizes.size() != staticL2AsTargetSizes.size()) {
-    dmaTransposeOnSource = false;
-  }
-
   OpFoldResult zeroVal = getAsIndexOpFoldResult(context, 0);
   OpFoldResult oneVal = getAsIndexOpFoldResult(context, 1);
 
-  if (dmaTransposeOnSource) {
+  FailureOr<bool> dmaTransposeOnTarget = isL2DmaTransposed(l3ToL2DmaOp, true);
+  if (failed(dmaTransposeOnTarget)) return failure();
+
+  if (!dmaTransposeOnTarget.value()) {
```

Comment:
The changes in this function are refactorring changes for the matmul + elementwise and I think they should be put in a separate PR to make it easier to review the splitting additions needed for 4x4 in this PR.

Reply:
The original change for matmul-elementwise was introduced in this PR https://github.com/nod-ai/iree-amd-aie/pull/850. This utility is also used in 4x4 matmul splitting, so here is just small refactor.

Reply:
It's not that small as there's quite a few changes in the top part of this file for it and it's completely independent, so I think it's really better for clarity/review purposes to put it in a separate PR to go in first.

Reply:
Okay, moved the refactor to a separate PR https://github.com/nod-ai/iree-amd-aie/pull/910.